### PR TITLE
fileURI() Pebble Function

### PR DESCRIPTION
**Pull request by asharan2buff**
### What changes are being made and why?

--- Crated a fileURI() to retrieve namespace from path.

### How the changes have been QAed?

---

### CODE REVIEW

File: core/src/main/java/io/kestra/core/runners/pebble/functions/GetFileURI.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import io.pebbletemplates.pebble.error.PebbleException;
+import io.pebbletemplates.pebble.extension.Function;
+import io.pebbletemplates.pebble.template.EvaluationContext;
+import io.pebbletemplates.pebble.template.PebbleTemplate;
+
+import java.util.List;
+import java.util.Map;
+
+public class GetFileURI implements Function {
+    private static final String ERROR_MESSAGE = "The 'getFileURI' function expects an argument 'path' that is a path to a namespace file or an internal storage URI.";
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/runners/pebble/functions/GetFileURI.java]:
```suggestion
    private static final String ERROR_MESSAGE = "The 'getFileURI' function expects an argument 'path' that is a path to a namespace file.";
```

File: core/src/main/java/io/kestra/core/runners/pebble/functions/GetFileURI.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    @Override
+    public List<String> getArgumentNames() {
+        return List.of("flow");
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/runners/pebble/functions/GetFileURI.java]:
```suggestion
        return List.of("path");
```

**Review by loicmathieu — CHANGES_REQUESTED** [review]
It also needs a test

**Comment by asharan2buff** [comment]
@loicmathieu I am unable to get the value of "flow" variable from context. I see a similar implementation in other functions. Am I missing anything?

**Comment by loicmathieu** [comment]
@asharan2buff it's because in test you will have a mocked run context with no variables in it.
We usually init variables manually and pass them to the render function as an extra parameter:

```java
Map<String, Object> variables = Map.of(
            "flow", Map.of(
                "id", "myFlow",
                "namespace", "myNamespace)
        );
variableRenderer.render("[...]", variables);
```

**Comment by asharan2buff** [comment]
@loicmathieu Added the test case. Please proceed with the review.

**Review by loicmathieu — APPROVED** [review]
LGTM, thanks

**Comment by MilosPaunovic** [comment]
Just a small note for the future @asharan2buff, if you could use **GitHub Keywords** when opening PRs to connect it to the proper issue (to automate the entire process a bit more), it would be super helpful to us. You can read a bit more about it [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue) (I've added closing keyword for this particular issue).

Thanks for this contribution! 🚀

**Pull request merged**